src/Microsoft.Health.Checkpoints/Storage/StorageCheckpointClient.cs

```diff
@@ -0,0 +1,143 @@
+        {
+            EnsureArg.IsNotNull(containerClient);
+
+            _storageClient = containerClient;
+            _logger = logger;
+        }
+
+        public virtual async Task<ICheckpoint> GetCheckpointAsync(string partition, string checkpointIdentifier)
+        {
+            EnsureArg.IsNotNullOrEmpty(partition, nameof(partition));
+            EnsureArg.IsNotNullOrEmpty(checkpointIdentifier, nameof(checkpointIdentifier));
+
+            ICheckpoint checkpoint = new Checkpoint();
+
+            try
+            {
+                var resultSegment = _storageClient.GetBlobsAsync(
+                                        traits: BlobTraits.Metadata,
+                                        states: BlobStates.All,
+                                        prefix: $"{partition}/{checkpointIdentifier}",
+                                        cancellationToken: CancellationToken.None)
+                                        .AsPages();
+
+                await foreach (Page<BlobItem> blobPage in resultSegment)
+                {
+                    if (blobPage.Values.Count == 0)
+                    {
+                        _logger.LogWarning($"No blob found for identifier {checkpointIdentifier}");
+                    }
+
+                    if (blobPage.Values.Count > 1)
+                    {
+                        _logger.LogWarning($"Multiple blobs found for identifier {checkpointIdentifier}");
+                    }
+
+                    foreach (BlobItem blobItem in blobPage.Values)
+                    {
+                        DateTimeOffset lastEventTimestamp = DateTime.MinValue;
+
+                        if (blobItem.Metadata.TryGetValue(_lastProcessedDateTime, out var dt))
+                        {
+                            DateTimeOffset.TryParse(dt, null, DateTimeStyles.AssumeUniversal, out lastEventTimestamp);
+                            checkpoint.LastProcessedDateTime = lastEventTimestamp;
+                        }
+
+                        if (blobItem.Metadata.TryGetValue(_lastProcessedIdentifier, out var id))
+                        {
+                            checkpoint.LastProcessedIdentifier = id;
+                        }
+
+                        if (checkpoint.LastProcessedDateTime == DateTime.MinValue || checkpoint.LastProcessedIdentifier == null)
+                        {
+                            _logger.LogWarning($"No valid checkpoint found for {checkpointIdentifier}. Using default checkpoint of {checkpoint.LastProcessedDateTime}");
+                        }
+
+                        checkpoint.ETag = blobItem.Properties.ETag.ToString();
+                        checkpoint.Identifier = checkpointIdentifier;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, ex.Message);
```

Comment:
Is this needed? Generally wouldn't this would be captured up the stack also and logged?

Reply:
No, not really. It can be captured elsewhere. Removed